pragma-sdk/pragma_sdk/common/fetchers/generic_fetchers/lp_fetcher/lp_contract.py

```diff
@@ -0,0 +1,98 @@
+from typing import Optional, Tuple, cast
+
+from starknet_py.net.full_node_client import FullNodeClient
+
+from pragma_sdk.common.logging import get_pragma_sdk_logger
+from pragma_sdk.common.types.types import Address
+from pragma_sdk.onchain.types import Contract
+from pragma_sdk.onchain.abis.abi import ABIS, get_erc20_abi
+
+logger = get_pragma_sdk_logger()
+
+Reserves = Tuple[int, int]
+
+
+class LpContract:
+    contract: Contract
+    _token_0: Optional[Contract] = None
+    _token_1: Optional[Contract] = None
+    _decimals: Optional[int] = None
+
+    def __init__(self, client: FullNodeClient, lp_address: Address):
+        self.contract = Contract(
+            address=lp_address,
+            abi=ABIS["pragma_Pool"],
+            provider=client,
+            cairo_version=1,
+        )
+
+    async def is_valid(self) -> bool:
+        """
+        Validate that the current contract is valid and usable.
+        We query every functions to make sure that it exists on-chain.
+        """
+        try:
+            _ = await self.get_reserves()
+            _ = await self.get_total_supply()
+            _ = await self.get_decimals()
+            _ = await self.get_token_0()
+            _ = await self.get_token_1()
```

Comment:
that's a lot of rpc calls tbh, wdyt that just calling the get_reserves is enough ? otherwise we can leverage on-chain read multicall

Reply:
did a multicall